### DISCUSSION TITLE
Template returned undefined, when trying to read a value from json file based on the key I give.

### DISCUSSION THREAD

**Discussion by tyehuoPHD** [SilentVoid13/Templater#1065] [Help] [answered]
# my problem and ask.
**problem** : Template returned undefined, when trying to read a value from json file based on the key I give.
**situation summary** : I managed to get the code to work when it's in a html file, but unable to make it work using obsidian.
**ask** : how to revise the code ?

# here's the .js file

```javascript
// key : selected value by user, supposed to correspond to some value on the first column of the json file
// value : the value on the second column of the json file
function fetchData(key) {
  try {
    const response = fetch(
      "D:/BaiduSyncdisk/ObsidianFolder/test/Templates/Scripts/dict_stock_code.json"
    );
    const data = response.json();
    const value = data[key];
    console.log(value);
    return temp;
  } catch (error) {
    console.error(error);
  }
}
module.exports = fetchData;

```
# here's my template
```bash
<% tp.user.fetchData("some_key") %>
```

# when i insert the template to a newly created note
```
undefined
```

# when i use the code in a html file, the output is fine. Here's the html code.
```html
<!DOCTYPE html>
<html>
  <head>
    <title>Read JSON File</title>
  </head>
  <body>
    <div id="output"></div>

    <script>
      // read the JSON file
      fetch("dict_stock_code.json")
        .then((response) => response.json())
        .then((data) => {
          const temp = data["中国联通"];
          // display the JSON data in the output div
          document.getElementById("output").innerHTML = temp;
        })
        .catch((error) => console.error(error));
    </script>
  </body>
</html>

```

**Comment by Zachatoo** [comment, marked as answer]
The code is different between your two scripts, so they may not work exactly the same.

The issue is likely that `fetch` returns a `Promise` object, which you either have to `await` or chain a `.then()` onto. You have done this in your html file, you have not done that in your templater script.

**Reply by tyehuoPHD** [reply]
thanks for the help
